# Exploratory Data Analysis
In this script we wish to explore the dataset and analysis on the features

In [5]:
# Import useful packages
%matplotlib inline
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
%load_ext autoreload
%autoreload 2

In [13]:
# Import custom functions
from scripts.proj1_helpers import *
from scripts.clean_data import *


In [11]:
data_path = 'data/train.csv'
y, x, ids = load_csv_data(data_path, sub_sample=False)

print('Size of x:', x.shape )

Size of x: (250000, 30)


In [12]:
print(x)

[[ 138.47    51.655   97.827 ...    1.24    -2.475  113.497]
 [ 160.937   68.768  103.235 ... -999.    -999.      46.226]
 [-999.     162.172  125.953 ... -999.    -999.      44.251]
 ...
 [ 105.457   60.526   75.839 ... -999.    -999.      41.992]
 [  94.951   19.362   68.812 ... -999.    -999.       0.   ]
 [-999.      72.756   70.831 ... -999.    -999.       0.   ]]


1. First off all we that we have a lot of values that are set to -999, which can be regarded as undefined.
2. By looking at the data we can see that the feature PRI_jet_num is a categorical feature, and have values 0-3. All others are continious.

To study the distribution of each feature, we plot the distribution of each feature. 

In [14]:
## MAYBE TODO: Plot distribution

In [15]:
# MAYBE TODO: DO PRINCIPLE COMPONENT ANALYSIS